src/commonMain/kotlin/fr/acinq/lightning/wire/OfferTypes.kt

```diff
@@ -714,15 +714,14 @@ object OfferTypes {
         } else {
             null // TODO: add exchange rates
         }
-        val description: String = records.get<OfferDescription>()!!.description
+        val description: String? = records.get<OfferDescription>()?.description
         val features: Features = records.get<OfferFeatures>()?.features ?: Features.empty
         val expirySeconds: Long? = records.get<OfferAbsoluteExpiry>()?.absoluteExpirySeconds
         private val paths: List<ContactInfo.BlindedPath>? = records.get<OfferPaths>()?.paths
         val issuer: String? = records.get<OfferIssuer>()?.issuer
         val quantityMax: Long? = records.get<OfferQuantityMax>()?.max?.let { if (it == 0L) Long.MAX_VALUE else it }
-        val nodeId: PublicKey = records.get<OfferNodeId>()!!.publicKey
-
-        val contactInfos: List<ContactInfo> = paths ?: listOf(ContactInfo.RecipientNodeId(nodeId))
+        val nodeId: PublicKey? = records.get<OfferNodeId>()?.publicKey
+        val contactInfos: List<ContactInfo> = paths ?: listOfNotNull(nodeId?.let { ContactInfo.RecipientNodeId(it) })
```

Comment:
```suggestion
        val contactInfos: List<ContactInfo> = paths ?: listOf(ContactInfo.RecipientNodeId(nodeId!!))
```

Reply:
Good idea, that makes it clear that `contactInfos` cannot be empty. Done in https://github.com/ACINQ/lightning-kmp/pull/626/commits/43d40efc3c0f89a51645c5f970554072b1547347